paas/install_client.go

```diff
@@ -485,15 +485,28 @@ func (c *InstallClient) fillInMissingSystemDomain(domain *kubernetes.Installatio
 		}
 
 		if ip != "" {
-			domain.Value = fmt.Sprintf("%s.nip.io", ip)
+			domain.Value = ip
+			if !strings.Contains(ip, ".nip.io") {
+				domain.Value = fmt.Sprintf("%s.nip.io", ip)
+			}
```

Comment:
Something is off here: the variable name is `ip`, the function name is `fetchIP`, but here you expect something other than an IP address to be returned/stored there.
The other thing that doesn't add up is the fact that a custom domain doesn't even have to be an IP address (i.e. it can be a regular domain name, without `nip.io` as a component). This means that your code needs to be more generic.

There's a corner case that I think you should consider, even though I would understand if you don't want to address it here: what if the user simply wants to re-run a FuseML installation and use a custom or default domain name that is different from the last installation ?

Could you please change the code to handle the discovery of an existing domain name outside of the previous `fetchIP` function, which serves a different purpose ? And expect the discovered domain name to be any generic domain name rather than a nip.io domain.

Reply:
Please take another look now